src/cli/cli.cpp

```diff
@@ -3203,6 +3203,39 @@ template <> otError Interpreter::Process<Cmd("dns")>(Arg aArgs[])
                                                         &Interpreter::HandleDnsServiceResponse, this, config));
         error = OT_ERROR_PENDING;
     }
+    /**
+     * @cli dns servicehost
+     * @cparam dns servicehost @ca{service-instance-label} @ca{service-name} <!--
+     * -->                 [@ca{DNS-server-IP}] [@ca{DNS-server-port}] <!--
+     * -->                 [@ca{response-timeout-ms}] [@ca{max-tx-attempts}] <!--
+     * -->                 [@ca{recursion-desired-boolean}]
+     * @par api_copy
+     * #otDnsClientResolveServiceAndHostAddress
+     * @par
+     * Send a service instance resolution DNS query for a given service instance
+     * with potential follow-up address resolution for its discovered host name.
+     * Service instance label is provided first, followed by the service name
+     * (note that service instance label can contain dot '.' character).
+     * @par
+     * The parameters after `service-name` are optional. Any unspecified (or zero)
+     * value for these optional parameters is replaced by the value from the
+     * current default config (`dns config`).
+     * @par
+     * Note: The DNS server IP can be an IPv4 address, which will be synthesized
+     * to an IPv6 address using the preferred NAT64 prefix from the network data.
+     * The command will return `InvalidState` when the DNS * server IP is an IPv4
+     * address but the preferred NAT64 prefix is unavailable.
+     * 'OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE' is required.
```

Comment:
I noticed that back ticks are not used here and similarly in other places within this source file. Was that intentional?
```suggestion
     * `OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE` is required.
```

Reply:
Not really. I just copied the documentation from the existing command which was using this. 
I fixed the similar ones is PR (we may want to update for all other cases in a seperate PR?).

Reply:
SGTM.